### PR TITLE
[feature] Popup Tooltip Widget

### PR DESCRIPTION
**Pull request by ruialmeida51**
Adds a customizable-ish popup widget to skeletoid.

Usage would be something like this:

```kotlin
        val myCustomPopupTooltip = PopupTooltip(
                anchorView = anchorView,
                message = "This is my message!",
                layout = R.layout.tooltip_layout,
                textViewId = R.id.tooltip_text,
                tooltipBackgroundId = R.id.tooltip_background
        )
                .arrow(true)
                .arrowStrokeColor(android.R.color.white)
                .layoutWidth(300)
                .verticalMargin(8)
                .verticalMarginClipped(16)
                .horizontalMargin(8)
                .horizontalMarginClipped(16)
//                .respectScreenBounds(true) //Use this when the clipped margins don't work well.
                .elevation(5)
                .touchDelegateTouchAreaIncreaseAmount(5)
                .arrowDrawable(R.drawable.tooltip_arrow_down)
                .arrowBackgroundId(R.id.tooltip_arrow)
                .arrowStrokeId(R.id.tooltip_arrow_stroke)
                .animationStyle(R.style.popup_window_animation)
                .arrowWidth(5)
                .arrowHeight(5)
                .indefinite(false)
                .timeToDismiss(2000L)
                .insideTouchDismissible(true)
                .outsideTouchDismissable(true)

        anchorView.setOnClickListener { myCustomPopupTooltip.show() }
```

### CODE REVIEW

**Comment by neteinstein** [comment]
@bruno-garces Don't we want to merge this?

**Comment by bruno-garces** [comment]
@neteinstein I thought this was already merged. I'll take a look thanks

**Comment by neteinstein** [comment]
@ruialmeida51 Can you add the info on how to use (example) to the README?

File: base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+        private val handler = Handler()
+
+        /**
+         * Clears the current reference to the tooltip and dismisses if it is present.
+         */
+        fun clearTooltipIfNotNull() {
+            tooltip?.apply {
+                if (isShowing) dismiss()
+                tooltip = null
+            }
```
Comment (bruno-garces) [inline base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt]: I find this hard to read. can we find a better idiomatic way to accomplish that

maybe
```
private fun PopupWindow.dismissIfShowing() {
    if (isShowing) dismiss()
}

fun clearTooltip() {
    tooltip?.dismissIfShowing()
    tooltip = null
}
```

File: base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+     */
+    fun horizontalMargin(margin: Int) = apply { this.horizontalMargin = margin }
+
+    /**
+     * The vertical offset from the anchor view to the tooltip when the bounds of the tooltip exceed
+     * the bounds of the screen.
+     *
+     * @param margin - the margin in dips.
+     */
+    fun verticalMarginClipped(margin: Int) = apply { this.verticalMarginClipped = margin }
```
Comment (bruno-garces) [inline base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt:151]: Just a curiosity, this apply block is here to allow chaining the setters like in a some sort of builder pattern?

Comment (ruialmeida51) [reply base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt:151]: Yup, that's exactly it 👍

File: base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt (hunk 1 above)
Comment (ruialmeida51) [reply base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt]: Sure, changed 👍

Comment (ruialmeida51) [reply base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt:120]: Sure 👍

Comment (neteinstein) [reply base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt]: @ruialmeida51 Please "resolve conversation" on all comments fixed ;-)

**Review by bruno-garces — APPROVED** [review]

**Comment by ruialmeida51** [comment]
> @ruialmeida51 Can you add the info on how to use (example) to the README?

Added :)

**Pull request merged**